userspace/libsinsp/sinsp.cpp

```diff
@@ -1268,9 +1268,15 @@ int32_t sinsp::next(OUT sinsp_evt **puevt)
 	m_parser->process_event(evt);
 #endif
 
-	// A side-effect of parsing this event may have generated a
-	// meta event. For example, parsing an execve or clone into a
-	// new cgroup may have created a container event.
+	if(!is_capture() && !m_meta_evt_pending)
+	{
+		m_container_manager.refresh();
```

Comment:
Since this could happen for every system call, I think it would be good to measure the performance. When there are outstanding requests, you'll try calling curl_multi_perform up to m_n_polling_attempts times. If each of those involves a select()/poll()/etc, that could amplify the number of system calls significantly.

If that is happening, you could use a run_on_interval to only check for responses every tens of ms or so.